src/main/webapp/components/services/utils.js

```diff
@@ -56,6 +56,58 @@
           return false;
         }
         return true;
+      },
+
+      setValue: function setValue(object, path, value) {
```

Comment:
I like to name this type of function `deepSet`.
In javascript, if you have object `a = {b: 3}` and do `a.c.d = 2` you will get `can not set property d of undefined` error because c does not exist. What you want sometimes is for that `c` to be automatically created as an empty object.
That is why we use `deepSet` function (`setValue` here), that creates missing objects on the way, so you can say `deepSet(a, 'b.c.d.e', 5)` and it will do this: `a.b.c.d.e = 5` while creating needed objects if missing.